lib/linguist/languages.yml

```diff
@@ -3738,12 +3738,12 @@ Ruby:
   - Deliverfile
   - Fastfile
   - Gemfile
-  - Gemfile.lock
```

Comment:
@brandonblack I don't follow your reasoning, in `File.write('hello.txt', 'Hello World')`, `hello.txt` is not a ruby file, files in `tmp/` or `log/` are not ruby files, `config/secrets.yml` is not a ruby, but all are associated with a rails project for instance.

e.g. in my use cases I use linguist as `Dir['*'].select { |file| is_ruby?(file) }.map { "do stuff on the ruby files" }` so I would like to avoid non-ruby files (e.g. `ruby Gemfile.lock` is a `syntax error` basically)